src/formatDate.js

```diff
@@ -8,7 +8,55 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newFormat = [];
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
```

Comment:
The assumption that the separator is always at index 3 in both `fromFormat` and `toFormat` arrays may not be correct. You should determine the separator dynamically or ensure that the format arrays are structured to include the separator explicitly.